shared-bindings/_bleio/__init__.c

```diff
@@ -78,6 +80,56 @@
 //|   This object is the sole instance of `_bleio.Adapter`.
 //|
 
+//| .. class:: BluetoothError(Exception)
+//|
+//|   Catch all exception for Bluetooth related errors.
+//|
+MP_DEFINE_EXCEPTION(BluetoothError, Exception)
```

Comment:
I think it makes sense for this to be a subclass of `OSError`, cf. subclasses of `OSError` here: https://docs.python.org/3/library/exceptions.html#os-exceptions

Reply:
I'd rather it not be because OSError usually do weird things with errnos. I took a look at urllib3 as a comparison and its exceptions are derived from Exception too. https://urllib3.readthedocs.io/en/latest/reference/index.html#urllib3.exceptions.HTTPError

Reply:
OK by me.